Academic year

Jay459 opened the pull request.

Jay459 committed:
controllers of academic year

Jay459 committed:
entities

Jay459 committed:
migrations of academic year

Jay459 committed:
academic year

Jay459 committed:
services

Jay459 committed:
archiving the data
--> archived the data from the original tables

Jay459 committed:
archiving tables

priyankcommits commented on src/entities/examination_archive.ts:
```diff
@@ -0,0 +1,30 @@
+import { AcademicYear } from "./academicYear";
+
+@Entity('Examinations_Archive')
+export class Examinations_Archive extends Timestamps {
```
it should be ExaminationsArchive

priyankcommits commented on src/entities/examination_archive.ts:
```diff
@@ -0,0 +1,30 @@
+    @Column({ nullable: true })
+    max_time: number
+
+    @ManyToOne(() => Teacher_Archive, (teacher_archive) => teacher_archive.examinations, { onDelete: "SET NULL", onUpdate: "CASCADE", nullable: true })
```
this should be TeacherArchive, class names should be pascal case

priyankcommits commented on src/routes/archive.ts:
```diff
@@ -0,0 +1,9 @@
+import Router from 'koa-router'
+const archiveRoutes = new Router()
+
+archiveRoutes.post('/archive/all', ArchiveControllers.archive);
```
this can be only "/archive"

priyankcommits commented on src/routes/archive.ts:
```diff
@@ -0,0 +1,9 @@
+
+archiveRoutes.post('/archive/all', ArchiveControllers.archive);
+
+archiveRoutes.get('/archive/getstudent', ArchiveControllers.getStudent)
```
its better to follow REST API standards, it should be /archive/student/get

priyankcommits commented on src/services/archive.ts:
```diff
@@ -0,0 +1,236 @@
+
+    public async archive(academicYearId) {
+
+        this.moveClass(academicYearId)
```
why not await everything so it goes in a sequence as intended instead of jumping ?

Jay459 committed:
changes made
--> followed the pascal scale in naming convention
--> corrected the migrations
--> followed the correct indentation
--> followed the rest api standards

Jay459 committed:
modifications done

Jay459 committed:
getting values depending upon academic year
--> getting the data of a particular academic year done
--> student_subject_archive table is done
--> intentation done

Jay459 committed:
migrations done

priyankcommits commented on src/controllers/academicYear.ts:
```diff
@@ -0,0 +1,29 @@
+import { Context } from 'koa';
+import { AcademicYearServices } from '../services/academicYear'
+
+export default class AcademicYearControllers {
```
Services and Controllers should not be plural. they should be Service and Controller
many line spaces, remove where not needed

priyankcommits commented on src/controllers/subject.ts:
```diff
@@ -33,13 +33,29 @@ export default class SubjectControllers {
 
         try {
-            const subjects = await subjectService.getSubjects()
+            const academicYearId = ctx.request.body
```
ctx.request.body contains data from client, hence it should be name data

priyankcommits commented on src/services/archive.ts:
```diff
@@ -0,0 +1,252 @@
+    }
+
+    public async removeNormalTables() {
+        const classes = await this.schoolClassRepository.find()
```
this is making two database calls, you can remove by passing the criteria to remove using the options, no need to first get them and then delete them.

Jay459 committed:
modification done
